File: 백준/Gold/2812. 크게 만들기/크게 만들기.py
N, K = map(int, input().split())
num = input()
s = []
for i in num:
	while (len(s) != 0 and K > 0):
		if (s[-1] < i):
			s.pop()
			K -= 1
		else:
			break;
	s.append(i)

while K > 0:
	s.pop()
	K -= 1
for i in s:
	print(i, end='')

File: 백준/Gold/2812. 크게 만들기/크게 만들기.cc
#include <bits/stdc++.h>
using namespace std;
int N, K;
string num;
stack<int> s1;
stack<int> s2;
int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);

	cin >> N >> K;
	cin >> num;
	for (int i = 0; i < num.length(); i++) {
		int now = num[i] - '0';
		while (!s1.empty() && K > 0 && s1.top() < now) {
			s1.pop();
			K--;
		}
		s1.push(now);
	}
	while (K > 0) {
		s1.pop();
		K--;
	}
	while (!s1.empty()) {
		s2.push(s1.top());
		s1.pop();
	}
	while (!s2.empty()) {
		cout << s2.top();
		s2.pop();
	}
	cout << '\n';
}
